### PR TITLE
13707 - Unsaved changes prompt

### PR DESCRIPTION
**Pull request by ChasBoucher**
*Issue #:* /bcgov/entity#13707

*Description of changes:*

This pr is a result of too many merge issues when rebasing from main(I got too far behind in fetching/rebasing code)...
I will close pr:  https://github.com/bcgov/ppr/pull/990/commits/d06b72a32024799a5876a94684646a7a53accf1b

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by dimak1** [comment]
@ChasBoucher there are some conflicts, not sure why you are getting those... are sure you pull/fetch the latest `main` branch before branching out to do your features/fixes?

**Comment by dimak1** [comment]
@ChasBoucher please update the PR title and description to tell details, is this a feature or fix, etc.

File: ppr-ui/src/components/common/ButtonFooter.vue [hunk 1]
```diff
@@ -199,7 +199,7 @@ export default defineComponent({
       )
     })
     const cancel = () => {
-      if (hasUnsavedChanges.value) localState.showCancelDialog = true
+      if (hasUnsavedChanges.value === true) localState.showCancelDialog = true
```
Comment (cameron-eyds) [inline ppr-ui/src/components/common/ButtonFooter.vue:202]: Could be simplified to `if (hasUnsavedChanges.value) ...`

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 2]
```diff
@@ -243,9 +255,20 @@ export default defineComponent({
     }
 
     const goToDash = (): void => {
-      context.root.$router.push({
-        name: RouteNames.DASHBOARD
-      })
+      if (hasUnsavedChanges.value === true) localState.showCancelDialog = true
```
Comment (cameron-eyds) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue:259]: Here as well

File: ppr-ui/src/views/newMhrRegistration/HomeOwners.vue [hunk 3]
```diff
@@ -214,15 +216,26 @@ export default defineComponent({
       }
     )
 
+    watch(
+      () => localState.showAddPersonSection,
+      () => {
+        if (isGlobalEditingMode.value === true) setUnsavedChanges(true)
```
Comment (cameron-eyds) [inline ppr-ui/src/views/newMhrRegistration/HomeOwners.vue:222]: Here as well

**Review by dimak1 — CHANGES_REQUESTED** [review]
There are some comments that require and update.

File: ppr-ui/src/components/common/ButtonFooter.vue (hunk 1 above)
Comment (ChasBoucher) [reply ppr-ui/src/components/common/ButtonFooter.vue:202]: I tested and it does not evaluate correct.  if not testing for true, the expression will not evaluate correct.  I have tested

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 2 above)
Comment (ChasBoucher) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue:259]: Same comment as above...

File: ppr-ui/src/views/newMhrRegistration/HomeOwners.vue (hunk 3 above)
Comment (ChasBoucher) [reply ppr-ui/src/views/newMhrRegistration/HomeOwners.vue:222]: same as above..

**Comment by ChasBoucher** [comment]
I have updated comments.

**Review by cameron-eyds — APPROVED** [review]

**Review by dimak1 — APPROVED** [review]

**Pull request merged**